lib/IMAP/MessageMapper.php

```diff
@@ -241,6 +241,29 @@ public function addFlag(Horde_Imap_Client_Socket $client,
 		);
 	}
 
+	public function getRawMessage(Horde_Imap_Client_Socket $client,
+								string $mailbox,
+								int $id): string {
+		$query = new Horde_Imap_Client_Fetch_Query();
+		$query->uid();
+		$query->fullText();
+
+
+		$result = iterator_to_array($client->fetch($mailbox, $query, [
+			'ids' => new Horde_Imap_Client_Ids($id),
+		]), false);
+
+		$msg = array_map(function (Horde_Imap_Client_Data_Fetch $result) {
+			return $result->getFullMsg();
+		}, $result);
+
+		if (!empty($msg)) {
+			return $msg[0];
+		} else {
+			return "Error: Could not get message source";
```

Comment:
let's just return `null` when there is no result

you can just `return reset($msg)` to get the value of the first element in the array https://www.php.net/manual/en/function.reset.php